src/workloads/query/PlanSelection.yml

```diff
@@ -0,0 +1,91 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload was created to reproduce various plan selection issues.
```

Comment:
I believe we would be looking at throughput. The purpose of adding these workloads to Genny is to measure the overall impact of plan selection issues, and their resolution, on the user. All of these cases are known to choose the "worse" plan. But if we fix this plan selection issue, and our fix creates extra overhead that outweighs the time that would be saved by choosing the right plan, then it is not an improvement for the user. We would be alerted in this situation by a regression in one of these workloads, or even the lack of the expected perf improvement.

With regards to the number of documents - I was modeling this after the original jstests, but I've realized those tests were meant to verify the incorrect plan was chosen (by checking explain output) rather than recreate a performance issue, so more documents are likely needed. I will need more time to run these queries in a local environment and play around to repro the performance difference. I'm going to convert this to a draft PR for now and re-request review once I've had time to look at this. Thank you!